src/Entity/ConfiguredPluginState.php

```diff
@@ -19,7 +19,7 @@ class ConfiguredPluginState
     /** @var string */
     private $version;
 
-    /** @var array */
+    /** @var array|mixed[] */
```

Comment:
Is mixed really needed?

Reply:
Don't really know... You cannot define the type exactly, because the structure is complex. But if there is only "array" the codestyle checker could possible say this is not enough...

Reply:
Maybe then an own class is better?

Reply:
I already thought about it while implementing the Configuration structure, so I can say:

No, as the structure differs from Provider to Provider. In my opinion the expense to implement multiple classes that are used based on the chosen provider would be too much.
By now the system is already over-engineered. I don't think that this would help more than it costs.